runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/artifact/ArtifactStagingService.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.fnexecution.artifact;
+
+import org.apache.beam.runners.fnexecution.FnService;
+
+/** An implementation of the Beam Artifact Staging Service. */
+public interface ArtifactStagingService extends FnService {
```

Comment:
That is indeed intended.  There are 2 expected users: the client uploading artifacts during job submission, and the runner retrieving those artifacts for distribution to workers.  So once we upload the artifacts we need to communicate them to the runner somehow, and this is done through the createAccessor method.  The alternative was to a) create a separate interface for createAccessor(),and b) then create a third interface that extends both ArtifactStagingService and interface (a).  However, since getting access to the artifacts once they are staged is a reasonable expectation to put on the ArtifactStagingService, I chose to just add that method to the existing interface.

Reply:
I can't say I'm quite convinced yet. Artifacts are used by 2 kinds of clients in 2 completely different ways - nobody needs both interfaces, so there's no need for a combined interface, and its existence may interfere with future refactoring. Moreover, in this PR, we never call createAccessor(), so I would suggest to remove this class from the current PR and defer the issue until the one where it is used (possibly after an offline discussion).

Reply:
Really they are used by 3 clients in 3 completely different ways (the sdk client, the invoker, and the sdk harness), but I see your point.  I'm happy to remove it from the current PR and have a design review of it.

Reply:
TBC offline